### ISSUE TITLE
Which permissions are required for deleting a resource?

### ISSUE THREAD

**Issue by RubenVerborgh** [solid/solid-spec#195]
When deleting a resource R in a container C, do we require:
- Write permissions on R?
- Write permissions on C?
- Both?
- Something else?

**Comment by RubenVerborgh** [comment]
History in https://github.com/solid/web-access-control-spec/pull/48.

Summary:
- The draft spec currently says "Write permissions on R"
- In a discussion, we said we wanted to follow Linux, so NSS was changed to "Write permissions on C"
- However, during the PR (https://github.com/solid/web-access-control-spec/pull/48), an alternative suggestion of "Write permissions on R _and_ C" came up.

We need to decide whether we want Linux compatibility or not.

**Comment by csarven** [comment]
If LDP's behaviour is applied to R and C: Write permissions on R as well as C is required in order to remove the containment triple from C. This requirement can be derived from LDP's `DELETE`: https://www.w3.org/TR/ldp/#ldpc-del-contremovesconttriple

**Comment by kjetilk** [comment]
What if we end up with `rm -r` (#190)? Then, it would need to be write on all resources down to and including the container?

**Comment by csarven** [comment]
I'd like to clarify https://github.com/solid/solid-spec/issues/195#issuecomment-559619352 because of a subtle point related to this issue. With the given that containment triples are server-managed, while Write is required on R, Write is not explicitly required on C for the purpose of removing the containment triple. There is an implicit write operation on C based on LDP's interaction model (when R is removed from containment triples) but that burden is not placed on the requesting agent, hence their access to C can be seen as out of scope. However, as LDPR's lifecycle is dependent and controlled by the LDPC in which it has a binding relationship with, it can be expected that any changes to that binding should require permission to alter C.

What's argued is more of a design decision based on connecting some basic assumptions. AFAICT, requiring Write on C and R in order to delete R is compatible with *nix behaviour.

**Comment by csarven** [comment]
>What if we end up with rm -r (#190)? Then, it would need to be write on all resources down to and including the container?

If `rm -r` is analogous, then it'd be reasonable to consider whether `-f` is applied by default as well ie. no prompt.
[Aside:
* I'm not sure whether  --one-file-system applies to our design.
* `--preserve-root` is taken as given in that ACL document R of root C can't be deleted.
* `-d` is used provided that a C is considered to be "empty" if its auxillary resources like acl and possibly meta doesn't count towards it.

]

Having `-f` as the default would be the simplest design - not to imply ideal - as opposed to for example having a workflow which prompts the user per child resource that's encountered along the way. `-I` may be useful in that there is some sort of a signal given back to the client (eg. can't delete C or need to delete contained C because it contains R).

In any case, `rm -r` should be treated as atomic. On *nix, user is only prompted if need to ie. all other removes will succeed.

**Comment by kjetilk** [comment]
Yeah, I think the attractiveness of `rm -r` is atomicity, which is hard to achieve for multiple HTTP requests.

As for `-f`, just a wild thought, it is not so much the prompt that I think characterizes `-f` but that you can delete a file if you would have permission to do so if you changed the permissions on the file and you have permission to do so... Which wasn't particularly clear, I suppose, but the wild thought I had was, perhaps `acl:Control` has a role to play...?

**Comment by csarven** [comment]
By not requiring Write on container to delete a contained resource, agents are not granted more access than what they need. Granting the agent with Write on container allows them to alter the container description as well as potentially modify or delete other (but not all) resources in the container.
